### PR TITLE
fix: Mining Key Spam

### LINKED ISSUE #1995: mining issue
mining bug, you spam E when you mine and it will give you twice the stuff

https://medal.tv/games/gta-v/clips/12BlSGrJv7_jsr/d1337yX13Qu8?invite=cr-MSwxT1ksNTE2OTA0NTYs

### PR DESCRIPTION
**Pull request by Savage912-Official**
Just added a check when clicking E to fix the spamming E issue.

closes #1995

### CODE REVIEW

**Comment by minipunch** [comment]
Can u use pmc-callbacks instead plz

**Comment by Savage912-Official** [comment]
> Can u use pmc-callbacks instead plz

Done

File: resources/usa_mining/client/main.lua [hunk 1]
```diff
@@ -3,14 +3,18 @@ local MINING = {
     {x = -592.55, y = 2076.86, z = 131.37, radius = 30}
 }
 
-
 Citizen.CreateThread(function()
+    local CheckingItems = TriggerServerCallback {
```
Comment (minipunch) [inline resources/usa_mining/client/main.lua:7]: Don't think this is where this belongs lol

File: resources/usa_mining/server/server.lua [hunk 2]
```diff
@@ -13,14 +13,20 @@ local rare = {
     {name = "Crude Oil", type = "misc", price = 450, legality = "legal", quantity = 1, weight = 15.0},
 }
 
+local CheckingItems = false
```
Comment (minipunch) [inline resources/usa_mining/server/server.lua:16]:
```suggestion

```

File: resources/usa_mining/server/server.lua [hunk 3]
```diff
@@ -13,14 +13,20 @@ local rare = {
...
+
 RegisterServerEvent('mining:doesUserHaveCorrectItems')
 AddEventHandler('mining:doesUserHaveCorrectItems', function()
     local char = exports["usa-characters"]:GetCharacter(source)
+    CheckingItems = true -- Begin Check
```
Comment (minipunch) [inline resources/usa_mining/server/server.lua:21]:
```suggestion
```

File: resources/usa_mining/server/server.lua [hunk 4]
```diff
@@ -13,14 +13,20 @@ local rare = {
...
+
     if char.hasItem('Pick Axe') then
         TriggerClientEvent("mining:startMining", source)
     else
         TriggerClientEvent("usa:notify", source, "You do not have the sufficient tools to mine!")
     end
+
+    CheckingItems = false -- Check Complete
```
Comment (minipunch) [inline resources/usa_mining/server/server.lua:29]:
```suggestion
```

**Review by minipunch — CHANGES_REQUESTED** [review]

**Pull request closed without merging**